### PR TITLE
Address warnings in the build

### PR DESCRIPTION
**Pull request by kleewho**
* To address deprecation in most places not deprecated has been used. In those specific areas where we actually want to verify that our deprecated methods are working I added SUPPRESS annotations.
* ktlint debug has been disabled to reduce the amount of information produced
* .editorconfig file used the suggested by ktlint value

### CODE REVIEW

File: pubnub-users/src/integrationTest/kotlin/com/pubnub/user/UserIntegTest.kt [hunk 1]
```diff
@@ -39,7 +38,8 @@ class UserIntegTest {
 
     @BeforeEach
     fun setUp() {
-        val config = PNConfiguration(UserId("kotlin")).apply {
+        UserId("kotlin")
```
Comment (marcin-cebo) [inline pubnub-users/src/integrationTest/kotlin/com/pubnub/user/UserIntegTest.kt:41]: What is this line for?

File: src/main/kotlin/com/pubnub/api/endpoints/FetchMessages.kt [hunk 2]
```diff
@@ -103,8 +103,8 @@ class FetchMessages internal constructor(
             effectiveMax(page.limit, includeMessageActions, channels.size).toString()
         queryParams["include_uuid"] = includeUUID.toString()
 
-        page.start?.run { queryParams["start"] = this.toString().toLowerCase(Locale.US) }
-        page.end?.run { queryParams["end"] = this.toString().toLowerCase(Locale.US) }
+        page.start?.run { queryParams["start"] = this.toString().lowercase(Locale.US) }
```
Comment (marcin-cebo) [inline src/main/kotlin/com/pubnub/api/endpoints/FetchMessages.kt:106]: Why in some places we have:
Locale.getDefault() and here Locale.US ?

**Review by marcin-cebo — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/pubnub/api/endpoints/FetchMessages.kt (hunk 2 above)
Comment (kleewho) [reply src/main/kotlin/com/pubnub/api/endpoints/FetchMessages.kt:106]: That's a very good question and I don't know. It's not an issue for a start or an end, but I guess we should have the same locales in the code

File: pubnub-users/src/integrationTest/kotlin/com/pubnub/user/UserIntegTest.kt (hunk 1 above)
Comment (kleewho) [reply pubnub-users/src/integrationTest/kotlin/com/pubnub/user/UserIntegTest.kt:41]: That's a leftover after failed attempt of mass fix

File: pubnub-spaces/src/integrationTest/kotlin/com/pubnub/space/SpaceIntegTest.kt [hunk 3]
```diff
@@ -38,7 +37,8 @@ class SpaceIntegTest {
 
     @BeforeEach
     fun setUp() {
-        val config = PNConfiguration(UserId("kotlin")).apply {
+        UserId("kotlin")
```
[the remark this answers is not in the document]
Comment (kleewho) [reply pubnub-spaces/src/integrationTest/kotlin/com/pubnub/space/SpaceIntegTest.kt:40]: That's a leftover after failed attempt of mass fix

**Review by marcin-cebo — APPROVED** [review]

**Pull request merged**